server/channels/store/sqlstore/channel_store.go

```diff
@@ -1517,7 +1525,9 @@ func (s SqlChannelStore) GetByNames(teamId string, names []string, allowFromCach
 			visited[name] = struct{}{}
 			var cacheItem *model.Channel
 			if err := channelByNameCache.Get(teamId+name, &cacheItem); err == nil {
-				channels = append(channels, cacheItem)
+				if includeArchivedChannels || cacheItem.DeleteAt == 0 {
```

Comment:
shouldn't this logic be an && instead of an || ?

Reply:
In every case, if DeleteAt is 0 we want to add the channel to the list of found. And if `includeArchivedChannels` is `true`, we can ignore the value of `DeleteAt`. So `||` wins! lol